src/Parser/Deathknight/Unholy/Modules/Features/Apocalypse.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+
+class Apocalypse extends Analyzer {
+  static dependencies = {
+    enemies: Enemies,
+  };
+
+  // used to track how many stacks a target has
+  targets = {};
+
+  totalApocalypseCasts = 0;
+  apocalypseWoundsPopped = 0;
+  
+  //These three sections are for keeping track of the amount of Festering Wounds on the target
+  on_byPlayer_applydebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+	}
+  }
+
+  on_byPlayer_removedebuffstack(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		this.targets[encodeTargetString(event.targetID, event.targetInstance)] = event.stack;
+    }
+  }
+
+  on_byPlayer_removedebuff(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.FESTERING_WOUND.id){
+		this.targets[encodeTargetString(event.targetID, event.targetInstance)] = 0;
+	}
+  }
+
+  //Logic that both counts the amount of Apocalypse cast by the player, as well as the amount of wounds popped by those apocalypse.
+  on_byPlayer_cast(event){
+    const spellId = event.ability.guid;
+    if(spellId === SPELLS.APOCALYPSE.id){
+		this.totalApocalypseCasts+=1;
+		if(this.targets.hasOwnProperty(encodeTargetString(event.targetID, event.targetInstance))) {
+			const currentTargetWounds = this.targets[encodeTargetString(event.targetID, event.targetInstance)];
+			if(currentTargetWounds > 5){
+				this.apocalypseWoundsPopped=this.apocalypseWoundsPopped + 6;
+				console.log(this.apocalypseWoundsPopped);
```

Comment:
Log lines for normal behavior should not make it to production code. Easiest is to remove this line.

If you want to leave in these debug prints for later development, you can hide it behind a debug flag like `debug && console.log(this.apocalypseWoundsPopped);` along with `const debug = false` at the top of the class. Then when you're on dev environment and you need to debug, you simply change the flag to true in order to turn on logging for this class.